### PR TITLE
feat: unify linter settings across indicators and add incremental formatting

### PR DESCRIPTION
**Pull request by dshemetov**
### Description
Related to https://github.com/cmu-delphi/delphi-epidata/pull/1328, unifying linting settings across indicators and these two repos.

### Changelog
* Use a single `pyproject.toml` for all indicators and `delphi_utils`, removed all `.pylintrc`
  * Made sure all indicators pass with the new config
  * Inlined a few pylint ignores that were previously covered by a rule in a `.pylintrc`, instead of turning them into global rules
  * Some whitespace formatting in the process (on the files I touched)
  * The line-length max is now bumped up to 120 from 100
  * Sorted the dependencies in the `setup.py` files of each indicator while I was at it (ignored in git blame)
* Add `make format` command to apply formatting to the changes on your current branch
* Update the template files as well
* Add a new darker format job to CI
* Add python package caching to CI
* Add "Lint and Format" section to README

### CODE REVIEW

**Review by melange396 — APPROVED** [review]
looks pretty good!
do you wanna make the same changes to the "template" files too?
* https://github.com/cmu-delphi/covidcast-indicators/blob/main/_template_python/.pylintrc
* https://github.com/cmu-delphi/covidcast-indicators/blob/main/_template_python/Makefile
* https://github.com/cmu-delphi/covidcast-indicators/blob/main/_template_python/setup.py

**Comment by dshemetov** [comment]
This was approved a while ago, I'm just going to merge so the new pipeline code can benefit from these changes.

**Comment by melange396** [comment]
looks like there were a lot of recent (force-pushed) changes since the old review...  can you summarize them?

**Comment by dshemetov** [comment]
Sure, I just
* updated the darker pin from `darker[flynt]~=1.7.1` to `darker[isort]~=2.1.1` (we don't use the flynt functionality)
* updated the templates to contain `make format` and have the darker dependency
* update the README file with some linting and formatting instructions
* updated the python CI so it runs the darker lint (couldn't get it to work before)
